Lab1/source/multiphasesorting.cpp

```diff
@@ -0,0 +1,224 @@
+#include <iostream> 
+#include <fstream>
+#include <random>
+#include <string>
+#include <vector>
+#include <climits>
+
+using std::cout;
+using std::cin;
+using std::endl;
+using std::cerr;
+using std::ifstream;
+using std::ofstream;
+using std::vector;
+using std::string;
+
+const int endElement = -1;
+int L = 1;
+
+bool createFileWithRandomNumbers(const std::string &fileName, const int numbersCount, const int maxNumberValue) { 
+    if (numbersCount <= 0 || maxNumberValue <= 0) { 
+        return false;
+    }
+
+    std::ofstream outFile(fileName);
+    if (!outFile.is_open())
+        return false;
+
+    std::random_device rd;
+    std::mt19937 generator(rd());
+    std::uniform_int_distribution<int> distribution(0, maxNumberValue);
+    
+    for (int i = 0; i < numbersCount; i++) {
+        int randomNumber = distribution(generator);
+        outFile << randomNumber << "\n";
+    }
+
+    if (!outFile.good()) {
+        outFile.close();
+        return false;    
+    }
+
+    outFile.close();
+    return true;
+}
```

Comment:
close будет вызван автоматически в деструкторе.
```suggestion
    if (!outFile.good()) {
        return false;    
    }

    return true;
}
```